Modules/Quotas/src/Quotas.Application/Metrics/UsedFileStorageSpaceMetricCalculator.cs

```diff
@@ -0,0 +1,19 @@
+﻿using Backbone.Modules.Quotas.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Quotas.Domain;
+
+namespace Backbone.Modules.Quotas.Application.Metrics;
+public class UsedFileStorageSpaceMetricCalculator : IMetricCalculator
+{
+    private readonly IFilesRepository _filesRepository;
+
+    public UsedFileStorageSpaceMetricCalculator(IFilesRepository filesRepository)
+    {
+        _filesRepository = filesRepository;
+    }
+
+    public async Task<uint> CalculateUsage(DateTime from, DateTime to, string uploader, CancellationToken cancellationToken)
+    {
+        var usedSpace = await _filesRepository.AggregateUsedSpace(uploader, from, to, cancellationToken);
+        return Convert.ToUInt32(usedSpace / 1024 / 1024);
```

Comment:
You should probably add a test for the exception. Just so that nobody thinks "Oh, we can simplify this with a simple cast". :)